apisix/plugins/forward-auth.lua

```diff
@@ -106,16 +106,20 @@ function _M.access(conf, ctx)
         method = conf.request_method
     }
 
+    local httpc = http.new()
     if params.method == "POST" then
         params.body = core.request.get_body()
+        local client_body_reader, err = httpc:get_client_body_reader()
+        if not err then
+            params.body = client_body_reader
+        end
```

Comment:
That won't work  if the `Content-Length` header is missing as `client_body_reader` requires it to be provided.

Apparently, reading `req_body` implicitly fixes this problem.

Reply:
I agree with @Gallardot, it's ok to use `core.request.get_body()`  if the `content-length` is missing, the client should ensure the input is correct. 
But we can't check the `err` value of `local client_body_reader, err = httpc:get_client_body_reader()`, because the `err` will be `nil` if the `content-length` is missing. https://github.com/ledgetech/lua-resty-http/blob/master/lib/resty/http.lua#L999

```
local httpc = http.new()
if params.method == "POST" then
    local client_body_reader, err = httpc:get_client_body_reader()
    if client_body_reader then
        params.body = client_body_reader
    else
        params.body = core.request.get_body()
    end
end
```

Reply:
if we go this way other test cases would need to be updated. This might result in a breaking change for users 🤔

Reply:
my bad, the  request already carries the content-length and other necessary headers... we just need to make sure that these headers are also sent to the auth server